### ISSUE TITLE
gna closing soon

### ISSUE THREAD

**Issue by leonhedding** [wkhtmltopdf/wkhtmltopdf#3390]
On 31/01/2017 the hosting servers for GNA announced they are closing. Will you be looking for another place to host the binary files?

https://mail.gna.org/public/project/2016-11/msg00001.html

**Comment by ashkulz** [comment]
Yep, didn't hear of that tbh. Will have to see where to host it...

**Comment by ashkulz** [comment]
Switched to self-hosting (see e71be6511b3afebadf13704adee7c4ea54d259c4). The only disadvantage is that any changes to the website get reflected after 30 minutes. On the other hand, now HTTPS is enabled via LetsEncrypt.

**Comment by ashkulz** [comment]
Thanks for the heads-up, @leonhedding!

**Comment by ashkulz** [comment]
I couldn't login to the VPS I was using to host it this morning. Although it looked like a configuration issue (host key kept regenerating after every boot, unable to login to SSH) I couldn't rule out it being compromised and brought it down. What I've done till date:

- shut down the possibly compromised server (`us.wkhtmltopdf.org`)
- reverted the downloads back to gna.org (ed804268eb24d995d46251a34586edba265d13ab)
- created a new server (`de.wkhtmltopdf.org`) which just hosts the website ... using HSTS meant that I couldn't host it at GitHub Pages anymore
- used the VPS console for `us.wkhtmltopdf.org` to verify that the downloads folder matched with the files hosted at gna.org (using `sha1sum`)
- also verified SHA256 checksums for latest release against [the release notes](https://github.com/wkhtmltopdf/wkhtmltopdf/releases/tag/0.12.4)

In summary, I don't think that any downloads were tampered with. If you wish to check any file downloaded recently, please check it against the checksums hosted at gna.org beside the file you downloaded. I don't think I'll be continuing with the current provider, just to be on the safer side.

Unfortunately, a [bandwidth usage of 3TiB/month](http://stats.gna.org/download.gna.org/wkhtmltopdf/) makes the cost very high at regular providers. I'm thinking of reaching out to a few CDNs if they'd be willing to host the downloads.

**Comment by hanoii** [comment]
Why not use GitHub releasees? I saw this mentioned on #3241 and you said that not because of older releases, but there must be a way. Maybe falling back to github pages with an archive for older releases and keeping the github release for the new ones?

**Comment by ashkulz** [comment]
Because of HSTS I deployed earlier, can't use GitHub pages. Also, downloads are 2.5GB and would clutter up repository. It won't take more than a few days to finalize...

**Comment by hanoii** [comment]
But do you need HSTS on GitHub pages?  Anyway, was just wondering but you definitely know better. I just think that if you find a way to not rely on any provider and not have to setup the server yourself you would have one less things to worry about. There must be good alternatives for open source projects.

**Comment by ashkulz** [comment]
No, you don't. Unfortunately, once it is set people can't access HTTP ... so anyone who's visited the site in the last 5-6 days won't be able to visit the site **at all** and will have to clear the HSTS settings (which is unintuitive and not easy).

I agree on the not preferring to manage the downloads, but there aren't any good solutions when your project becomes moderately popular and has bandwidth needs in the TBs ... even [phantomjs has a similar issue](https://github.com/ariya/phantomjs/issues/13953).

**Comment by hanoii** [comment]
Just thinking out loud,  why not relying on GitHub releases for newer binaries and keeping the old ones in whichever hosting platform you find for the time being, with time the bandwidth requirement for older versions should decrease considerably.

Might want to reach out to digital ocean as well, I think they are supporting of open source projects.

And couldn't you go back eventually to github pages by sending out an HSTS header from your own server (host it temporarily) with a very log TTL?

**Comment by ashkulz** [comment]
Migrated to Scaleway VC1S in France (`fr.wkhtmltopdf.org`) which offers 200Mbit/s unmetered bandwidth and enabled in 5dfb88d1c1f93da14a1bab6b7223575220c5d882.

**Comment by bikalbasnet** [comment]
New server https://downloads.wkhtmltopdf.org/0.12/0.12.4/ seems to unreliable. Server is not responding properly. Could be bandwidth issue :cry:

**Comment by thin-k-design** [comment]
I can confirm, I'm receiving like 5 kb per minute from the server... Can't download anything :/

**Comment by AnghelMihaiStanciu** [comment]
the same here
earlier today not even the site was loading
now website is wroking fine but downloads are not

**Comment by dvershinin** [comment]
Self-hosting might be causing more spending than saving due to electricity bills. I suggest to go with Linode. If you need professional server setup with Linode and the likes of VPS providers - I can help :)

**Comment by ashkulz** [comment]
See #3518 -- releases are now hosted on Github.
